staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -527,6 +527,15 @@ func (cnc *CloudNodeController) getProviderID(ctx context.Context, node *v1.Node
 		return node.Spec.ProviderID, nil
 	}
 
+	if instancesV2, ok := cnc.cloud.InstancesV2(); ok {
+		metadata, err := instancesV2.InstanceMetadata(ctx, node)
```

Comment:
This change seems reasonable to me, would it be possible to add a unit test for this?

Reply:
@andrewsykim I rethink about this function, I still think we should return the providerID since on Line 403 it will need the providerID to update the node
if providerID for InstancesV2 is always empty, then `n.Spec.ProviderID` will not be updated to the correct one

Reply:
It seems like instead, `getNodeModifiersFromCloudProvider` needs to be updated to fill in `node.Spec.ProviderID` if `providerID == ""` but `instanceMeta.ProviderID != ""`.